### PR TITLE
lf-pico-template updates

### PR DESCRIPTION
**Pull request by gundralaa**
Added minimal changes to allow for RP2040 target.
- [x] default C target backend to LFC
- [x] rp2040 init using template

note: i've addressed comments in https://github.com/lf-lang/lingo/pull/39. please merge that into this branch first.

### CODE REVIEW

**Comment by lhstrh** [comment]
This looks good! Is it ready for review?

File: src/package/mod.rs [hunk 1]
```diff
@@ -209,31 +210,40 @@ impl ConfigFile {
...
-
         // Remove .git, .gitignore ad temporary folder
         remove_file(".gitignore")?;
         remove_dir_all(Path::new(".git"))?;
         remove_dir_all(tmp_path)?;
         Ok(())
     }
 
+    // Sets up a LF project with Zephyr as the target platform.
+    pub fn setup_zephyr(&self) -> BuildResult {
```
Comment (oowekyala) [inline src/package/mod.rs:229]: these don't need to be public

File: src/package/mod.rs [hunk 2]
```diff
@@ -209,31 +210,40 @@ impl ConfigFile {
...
+    // Sets up a LF project with RP2040 MCU as the target platform.
+    // Initializes a repo using the lf-pico-template
+    pub fn setup_rp2040(&self) -> BuildResult {
+        let url = "https://github.com/lf-lang/lf-pico-template";
+        self.setup_template_repo(url)
+    }
+
     pub fn setup_example(&self) -> BuildResult {
         if is_valid_location_for_project(Path::new(".")) {
             match self.apps[0].platform {
```
Comment (oowekyala) [inline src/package/mod.rs:243]: Why is that only done for the first app? What if there are no declared apps?

**Comment by revol-xut** [comment]
@gundralaa Can u finish this up ?

File: src/package/mod.rs (hunk 2 above)
Comment (gundralaa) [reply src/package/mod.rs:243]: seems that the init command creates a config file with one app that reflects the arguments passed (platform and language).  an invalid config file from the init command should be caught earlier on. init should only operate one app.

File: src/package/mod.rs (hunk 1 above)
Comment (gundralaa) [reply src/package/mod.rs:229]: addressed
https://github.com/lf-lang/lingo/pull/39

File: src/package/mod.rs [hunk 3]
```diff
@@ -209,31 +210,40 @@ impl ConfigFile {
         Ok(())
     }
 
-    // Sets up a LF project with Zephyr as the target platform.
-    pub fn setup_zephyr(&self) -> BuildResult {
-        // Clone lf-west-template into a temporary directory
-        let tmp_path = Path::new("zephyr_tmp");
+    pub fn setup_template_repo(&self, url: &str) -> BuildResult {
+        let tmp_path = Path::new("tmp");
```
[the remark this answers is not in the document]
Comment (gundralaa) [reply src/package/mod.rs:214]: addressed
https://github.com/lf-lang/lingo/pull/39

**Comment by lhstrh** [comment]
@gundralaa, there is a conflict with what's in `main`. Could you take a quick look? Thanks!

**Comment by gundralaa** [comment]
https://github.com/lf-lang/lingo/pull/41 fixed the conflict here

**Comment by revol-xut** [comment]
@gundralaa Did u test your changes can u produce binaries that can run on the pico ?

**Comment by revol-xut** [comment]
From the code side of things this looks very good!

**Review by lhstrh — APPROVED** [review]
LGTM!

**Comment by gundralaa** [comment]
> @gundralaa Did u test your changes can u produce binaries that can run on the pico ?

Yes, it works with the current Lingo.toml in lf-pico-template. Binaries are generated in the target directory.

**Review by revol-xut — APPROVED** [review]

**Pull request merged**